RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -982,14 +1270,28 @@ class DeepTauId : public deep_tau::DeepTauBase {
     event.getByToken(rho_token_, rho);
 
     tensorflow::Tensor predictions(tensorflow::DT_FLOAT, {static_cast<int>(taus->size()), deep_tau::NumberOfOutputs});
+
+
     for (size_t tau_index = 0; tau_index < taus->size(); ++tau_index) {
+
       std::vector<tensorflow::Tensor> pred_vector;
-      if (version == 1)
-        getPredictionsV1(taus->at(tau_index), *electrons, *muons, pred_vector);
-      else if (version == 2)
-        getPredictionsV2(taus->at(tau_index), *electrons, *muons, *pfCands, vertices->at(0), *rho, pred_vector);
-      else
-        throw cms::Exception("DeepTauId") << "version " << version << " is not supported.";
+      if (is_online){
+        if (version == 1)
+          getPredictionsV1<reco::PFCandidate, reco::PFTau>(taus->at(tau_index), tau_index, electrons, muons, pred_vector, tauIDs);
+        else if (version == 2){
+          getPredictionsV2<reco::PFCandidate, reco::PFTau>(taus->at(tau_index), tau_index, electrons, muons, *pfCands, vertices->at(0), *rho, pred_vector, tauIDs);
+        }
+        else
+          throw cms::Exception("DeepTauId") << "version " << version << " is not supported.";
+      } else {
+        if (version == 1)
+          getPredictionsV1<pat::PackedCandidate, pat::Tau>(taus->at(tau_index), tau_index, electrons, muons, pred_vector, tauIDs);
+        else if (version == 2)
+          getPredictionsV2<pat::PackedCandidate, pat::Tau>(taus->at(tau_index), tau_index, electrons, muons, *pfCands, vertices->at(0), *rho, pred_vector, tauIDs);
+        else
+          throw cms::Exception("DeepTauId") << "version " << version << " is not supported.";
+      }
```

Comment:
Maybe it would be better to put the if-statements the other way round in order to avoid the duplication of the exception and to have versions (also future ones) in a single place each.